File: packages/react-native-session-replay/android/src/main/java/com/ft/sdk/reactnative/sessionreplay/utils/ReactNativeUtils.java
package com.ft.sdk.reactnative.sessionreplay.utils;

public class ReactNativeUtils {

    private ReactNativeUtils() {
    }

    public static Integer convertToNativeInt(Object value) {
        if (value == null) {
            return null;
        }
        try {
            return (int) Double.parseDouble(value.toString());
        } catch (NumberFormatException e) {
            return null;
        }
    }
}

File: android/src/main/java/com/cloudcare/ft/mobile/sdk/tracker/reactnative/utils/ReactNativeUtils.kt
package com.cloudcare.ft.mobile.sdk.tracker.reactnative.utils

class ReactNativeUtils {

  companion object {
    /**
     * react native number 转化 int
     */
    fun convertToNativeInt(value: Any?): Int? {
      return value?.toString()?.toDoubleOrNull()?.toInt()
    }

    /**
     * react native number 转化 long
     */
    fun convertToNativeLong(value: Any?): Long? {
      return value?.toString()?.toDoubleOrNull()?.toLong()
    }
  }
}
